src/phoenix/server/api/mutations/playground_chat_mutations.py

```diff
@@ -0,0 +1,116 @@
+# mutations.py
+
+from collections.abc import Iterable, Iterator
+from typing import Any, Optional
+
+import strawberry
+from strawberry.types import Info
+from typing_extensions import TypeAlias, assert_never
+
+from phoenix.server.api.context import Context
+from phoenix.server.api.exceptions import BadRequest
+from phoenix.server.api.helpers.playground_clients import initialize_playground_clients
+from phoenix.server.api.helpers.playground_registry import PLAYGROUND_CLIENT_REGISTRY
+from phoenix.server.api.helpers.playground_spans import streaming_llm_span
+from phoenix.server.api.input_types.ChatCompletionInput import ChatCompletionInput
+from phoenix.server.api.input_types.TemplateOptions import TemplateOptions
+from phoenix.server.api.types.ChatCompletionMessageRole import ChatCompletionMessageRole
+from phoenix.server.api.types.ChatCompletionSubscriptionPayload import (
+    FinishedChatCompletion,
+)
+from phoenix.server.api.types.TemplateLanguage import TemplateLanguage
+from phoenix.server.dml_event import SpanInsertEvent
+from phoenix.utilities.template_formatters import (
+    FStringTemplateFormatter,
+    MustacheTemplateFormatter,
+    TemplateFormatter,
+)
+
+initialize_playground_clients()
+
+ChatCompletionMessage: TypeAlias = tuple[
+    ChatCompletionMessageRole, str, Optional[str], Optional[list[str]]
+]
+
+
+@strawberry.type
+class PlaygroundChatCompletionMutationMixin:
+    @strawberry.mutation
+    async def playground_chat_completion(
+        self, info: Info[Context, None], input: ChatCompletionInput
+    ) -> FinishedChatCompletion:
+        provider_key = input.model.provider_key
+        llm_client_class = PLAYGROUND_CLIENT_REGISTRY.get_client(provider_key, input.model.name)
+        if llm_client_class is None:
+            raise BadRequest(f"No LLM client registered for provider '{provider_key}'")
```

Comment:
can we get this returned as an error so it's easier to display in the front end, the FinishedChatCompletion type has an errors field right?

Reply:
`BadRequest` is a custom graphql error, so it should be displayable in the frontend

Reply:
Relay has an `onError` hook https://relay.dev/docs/api-reference/use-mutation/#return-value

Reply:
yeah it just requires a whole lot of parsing to get the real error from the relay error

Reply:
i see. probably a broader discussion if we want to add error types on mutations / query. we've avoided that pattern for the sake of simplicity so far.

Reply:
don't we have that on the streaming version of this though? an error variant in the union

Reply:
but yes don't need to use it here just saying it can be helpful for handling known errors more gracefully for us, I know it's been shared before but this is the inspiration https://sachee.medium.com/200-ok-error-handling-in-graphql-7ec869aec9bc https://productionreadygraphql.com/2020-08-01-guide-to-graphql-errors

Reply:
> don't we have that on the streaming version of this though? an error member of the union, i guess that is technically not mutation or query (subscription) but was thinking it would set the precedent here

the error types in the subscription were unavoidable because we need to continue yielding payloads (`FinishedChatCompletion` in particular) even after an error has occurred. We don't currently have the same pattern on mutations and queries afaik

Reply:
ahh okay, i didn't know that was why those were added, we can punt it for now, we need to parse errors from relay anyways for various other pages so will make a ticket to get that working